test/functorch/test_control_flow.py

```diff
@@ -431,6 +462,260 @@ def fwbw(map_op, f, x, y):
+
+        # Jax Examples
+        x = torch.arange(0, 4)
+        cumsum = associative_scan(add, x, 0, generic_scan=True)
+        cumsum_exp = _fake_associative_scan(add, x, 0)
+        self.assertEqual(cumsum, torch.tensor([0.0, 1.0, 3.0, 6.0], dtype=torch.int64))
+        self.assertEqual(cumsum, cumsum_exp)
+
+        cumsum = associative_scan(add, x, 0, generic_scan=True, reverse=True)
+        cumsum_exp = _fake_associative_scan(add, x, 0, reverse=True)
+        self.assertEqual(cumsum, torch.tensor([6.0, 6.0, 5.0, 3.0], dtype=torch.int64))
+        self.assertEqual(cumsum, cumsum_exp)
+        ######################
+
+        x = torch.randn(3, 2, 2, requires_grad=True)
+        gm = make_fx(f)(add, x, 0)
+        self.assertExpectedInline(
+            gm.print_readable(print_output=False).strip(),
+            """\
+class f(torch.nn.Module):
+    def forward(self, op_1, x_1: "f32[3, 2, 2]", dim_1):
+        # No stacktrace found for following nodes
+        _tensor_constant0 = self._tensor_constant0
+        _unsafe_view: "f32[3, 2, 2]" = torch.ops.aten._unsafe_view.default(_tensor_constant0, [3, 2, 2]);  _tensor_constant0 = None
+        return _unsafe_view""",
+        )
+
+        x = torch.randn(3, 2, 2, requires_grad=True)
+        for op, op_pt in [(add, torch.cumsum), (mul, torch.cumprod)]:
+            cumsum = associative_scan(op, x, 0, generic_scan=True)
+            cumsum_exp = _fake_associative_scan(op, x, 0)
+            self.assertEqual(cumsum, cumsum_exp)
+            cumsum_exp_PT = op_pt(x, 0)
+            self.assertEqual(cumsum, cumsum_exp_PT)
+
+            cumsum = associative_scan(op, x, 0, generic_scan=True)
+            cumsum_exp = _fake_associative_scan(op, x, 1)
+            self.assertNotEqual(cumsum, cumsum_exp)
+
+            cumsum = associative_scan(op, x, 1, generic_scan=True)
+            cumsum_exp = _fake_associative_scan(op, x, 1)
+            self.assertEqual(cumsum, cumsum_exp)
+
+        import random
+
+        num_dims = [random.randint(2, 5) for _ in range(5)]
+        for num_dim in num_dims:
+            shapes = [random.randint(1, 13) for _ in range(num_dim)]
+            x = torch.randn(*shapes, requires_grad=True)
+            for op, op_pt in [(add, torch.cumsum), (mul, torch.cumprod)]:
+                cumsum = associative_scan(op, x, 0, generic_scan=True)
+                cumsum_exp = _fake_associative_scan(op, x, 0)
+                self.assertEqual(cumsum, cumsum_exp)
+                cumsum_exp_PT = op_pt(x, 0)
+                self.assertEqual(cumsum, cumsum_exp_PT)
+
+    def test_associative_scan_host_side_autograd(self):
+        def add(x: torch.Tensor, y: torch.Tensor):
+            return x + y
+
+        x = torch.randn(3, 2, 2, requires_grad=True)
+
+        for direction in [False, True]:
+            result = associative_scan(add, x, 0, generic_scan=True, reverse=direction)
+            expexted_result = _fake_associative_scan(add, x, 0, reverse=direction)
```

Comment:
nit. expected
There are a few expexted in the test

Reply:
Done